### ISSUE TITLE
[NFR] New events to emitter

### ISSUE THREAD

**Issue by Zaszczyk** [thephpleague/oauth2-server#825]
There are already a few events:
-client.authentication.failed
-user.authentication.failed
-refresh_token.client.failed

I would like to propose add new events to emitter:
-refresh_token.issued
-access_token.issued

I will use it to log in my application generated new access_token, for example:
```
$server->getEmitter()->addListener(
        'access_token.issued',
        function (\League\OAuth2\Server\RequestEvent $event) {
            MyOwnLogger::add($event->getName(), new \DateTime());
        }
    );
```

**Comment by Sephster** [comment]
Hi @Zaszczyk - I've discussed this with some of the other maintainers and they haven't raised any issues so I think this is a reasonable proposal. Would you be willing/able to submit a PR for this?

**Comment by Sephster** [comment]
Hi @Zaszczyk, just wondering if you were willing to submit a PR? If not, I will try and pick this up but it won't be for a few months yet. Many thanks.

**Comment by Zaszczyk** [comment]
Hi @Sephster, I'm sorry for the lack of response, I had not resources and time to code it. However it is interesting challenge and I will try to pick this up until end of March ok?

**Comment by Sephster** [comment]
No pressure at all. That is fine. I don't have much time to pick it up myself but if a window comes about and you haven't picked it up yet, I will add this myself. Thanks for agreeing to help out

**Comment by Zaszczyk** [comment]
@Sephster are there tests of emitting events or already implemented events? Because I want to inspire, but can't find any.

**Comment by Sephster** [comment]
I don't believe there are any tests for these in our test suite.

**Comment by Sephster** [comment]
Implemented in commit  6fd3024
